pkg/client/cache/listers.go

```diff
@@ -265,6 +265,26 @@ func (s *StoreToReplicationControllerLister) GetPodControllers(pod *api.Pod) (co
 	return
 }
 
+type StoreToDeploymentNamespacer struct {
```

Comment:
You should just use an Indexer instead of a Store in the `StoreToDeploymentLister` and then provide a `storeDeploymentsNamespacer` to pick the namespace you want. Look for example how the `storeReplicationControllersNamespacer` is implemented.